packages/circuits/test/initial.test.js

```diff
@@ -0,0 +1,101 @@
+const circomlib = require('circomlibjs');
+const ff = require('ffjavascript');
+const { callGenWitness } = require('circom-helper');
+
+describe("Initial Hand Test", () => {
+    const circuit = 'initial.test';
+    let mimcsponge;
+    let deckLeaves = [], handLeaves = [];
+    let deckRoot;
+    let newDeckRoot, newHandRoot;
+
+    beforeAll(async () => {
+        mimcsponge = await circomlib.buildMimcSponge();
+
+        // initialize deck leaves and hand leaves
+        for (let i = 0; i < 64; i++) {
+            deckLeaves.push(BigInt(i));
+            handLeaves.push(BigInt(255));
+        }
+        deckRoot = getMerkleRoot(deckLeaves, mimcsponge);
+    });
+
+    // set longer timeout for test
+    jest.setTimeout(10000);
+    it("Should correctly construct a merkle tree root", async () => {
+        // assume user draws 7 cards
+        const maxDeckSize = 64;
+        let drawnIndices = [2,4,6,8,10,12,14];
+
+        // update hand leaves
+        let newHandLeaves = [...handLeaves];
+        for (let i = 0; i < drawnIndices.length; i++) {
+            newHandLeaves[i] = deckLeaves[drawnIndices[i]];
+        }
+        newHandRoot = getMerkleRoot(newHandLeaves, mimcsponge);
+
+        drawnIndices = new Set(drawnIndices);
+        let drawnCardIndices = Array(64).fill(0);
+        for (const index of drawnIndices) {
+            drawnCardIndices[index] = 1;
+        }
+        let deckPredicate = []
+        let deckCount = 0;
+        let handCount = 0;
+        for (let i = 0; i < 64; i++) {
+            if (drawnIndices.has(i)) {
+                // card is drawn
+                deckPredicate.push(maxDeckSize ** handCount);
+                handCount++;
+            } else {
+                // card is not drawn
+                deckPredicate.push(maxDeckSize ** deckCount);
+                deckCount++;
+            }
+        }
```

Comment:
It would be good to test some shuffling here.

Reply:
Added the shuffle test in another file :)

Reply:
It does test another circuit though (which is not reused by the `Initial` circuit) so it would be easily possible for the circuit not to check the shuffling (the predicates) correctly :p

Reply:
That makes sense, have made the changes.